docs/spec/src/integration/spec/5-lifecycle-phases.md

```diff
@@ -0,0 +1,169 @@
+    There are different variants of the attack. The basic invariant the system needs to satisfy is that with the chunks from sufficient set of validators, you can get back the full blob. So the total size of the chunks held by these validators needs to exceed the blob size. If I don't know the blob size (or at least an upper bound), there's no way for the system to validate this invariant.
+    Here’s a simple example. Assume a network of 8 DA nodes, and coding ratio 1/2. For a `blob` containing 128 field elements (FEs), each node gets 128*2/8=32 FEs, meaning that any 4 nodes can join forces and reconstruct the data. Now assume a world without length proof; a malicious disperser receives the same blob, uses the same commitment, but claims that the blob only had length 4 FEs. He sends each node 4*2/8=1 FE. The chunks submitted to the nodes match the commitment, so the nodes accept and sign over the blob’s batch. But now there are only 8 FEs in the system, which is not enough to reconstruct the original blob (need at least 128 for that).
+    
+
+> Note that the length here is the length of the blob (power of 2), which is different from the payload_length encoded as part of the `PayloadHeader` in the `blob` itself (see the [encoding section](#encoding)).
+> 
+
+**PaymentHeader**
+
+The paymentHeader specifies how the blob dispersal to the network will be paid for. There are 2 main modes of payment, the permissionless pay-per-blob model and the permissioned reserved-bandwidth approach. See the [Payments](https://docs.eigenda.xyz/releases/payments) release doc for full details; we will only describe how to set these 4 fields here.
+
+```protobuf
+message PaymentHeader {
+  // The account ID of the disperser client. This should be a hex-encoded string of the ECSDA public key
+  // corresponding to the key used by the client to sign the BlobHeader.
+  string account_id = 1;
+  uint32 reservation_period = 2;
+  bytes cumulative_payment = 3;
+    // The salt is used to ensure that the payment header is intentionally unique.
+  uint32 salt = 4;
+}
+```
+
+Users who want to pay-per-blob need to set the cumulative_payment. Users who have already paid for reserved-bandwidth can set the reservation_period.
+
+An rpc call to the Disperser’s `GetPaymentState` method can be made to query the current state of an `account_id`. A client can query for this information on startup, cache it and then update it manually when making pay-per-blob payments, by keeping track of its current `cumulative_payment` to set it correctly for subsequent dispersals.
+
+The `salt` is needed for reserved bandwidth users: given that the `blob_header_hash` uniquely identifies a dispersal, if a dispersal ever fails (see next section), then to redisperse the same blob, a new unique `blob_header` must be created. If the `reservation_period` is still the same (currently set to 300 second intervals), then the salt must be increased (or randomly changed) to allow resubmitting the same `blob`.
```

Comment:
This doc describes payment with "salt", but the latest iteration uses timestamp instead. Is this documentation copied from somewhere else, or did you generate it for this PR?

Reply:
ah good call. will update

Reply:
[176dd65](https://github.com/Layr-Labs/eigenda/pull/1439/commits/176dd652e57e14075b2e32d93af34488a4fff2d9)